File: cpp/2202/220224.cpp
#include <vector>
#include <algorithm>

/**
 * Definition for singly-linked list.
 */
struct ListNode {
	int val;
	ListNode* next;

	ListNode(int x = 0, ListNode* next = nullptr) : val(x), next(next) {}
};

/**
 * 148. Sort List
 * Given the head of a linked list, return the list after sorting it in ascending order.
 */
class Solution {
public:
	static ListNode* sortList(ListNode* head) {
		if (!head)
			return nullptr;
		std::vector<ListNode*> arr;
		for (ListNode* ptr = head; ptr; ptr = ptr->next)
			arr.push_back(ptr);
		std::sort(arr.begin(), arr.end(), [](ListNode* x, ListNode* y){ return x->val < y->val; });
		int n = arr.size();
		for (int i = 0; i < n - 1; ++i)
			arr[i]->next = arr[1 + i];
		arr.back()->next = nullptr;
		return arr[0];
	}
};

int main() {
	return 0;
}

File: old/golang/2202/220224.go
package main

import "sort"

/**
 * Definition for singly-linked list.
 */

type ListNode struct {
	Val  int
	Next *ListNode
}

func sortList(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}

	arr := make([]*ListNode, 0)
	for ptr := head; ptr != nil; ptr = ptr.Next {
		arr = append(arr, ptr)
	}

	sort.Slice(arr, func(i, j int) bool {
		return arr[i].Val < arr[j].Val
	})

	n := len(arr)
	arr[n-1].Next = nil
	for i := 0; i < n-1; i++ {
		arr[i].Next = arr[1+i]
	}

	return arr[0]
}

func main() {

}
